### PR TITLE
Rsa golang wrappers

### PR DESCRIPTION
**Pull request by AlessandroBudroni**
- Add  GO Wrappers to support RSA
- Add self-tests (I will create more)
- Re-organized all the wrappers in different files: amcl.go, rsa.go, mpin.go

### CODE REVIEW

File: wrappers/go/rsa.go.in [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#include <stdlib.h>
+#include "amcl.h"
+#include "randapi.h"
+#include "rsa.h"
+#include "utils.h"
+*/
+import "C"
+
+// generate an RSA key pair
+func RSA_KeyPair(RNG *RandNG, e int32, P []byte, Q []byte) (C.rsa_private_key, C.rsa_public_key) {
```
Comment (VMitov) [inline wrappers/go/rsa.go.in]: Can't we loose the underscore.

File: wrappers/go/rsa.go.in [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	defer OctetFree(&COct)
+
+	rtn := C.PKCS15(C.int(hashType), &MOct, &COct)
+	errorCode = int(rtn)
+	C = OctetToBytes(&COct)
+	return errorCode, C[:]
+}
+
+// OAEP Message Encoding for Encryption
+func OAEP_ENCODE(hashType int, M []byte, RNG *RandNG, P []byte) (errorCode int, F []byte) {
```
Comment (VMitov) [inline wrappers/go/rsa.go.in]: Can we use the golang convention for naming functions - e.g. for this particular function OAEPEncode?

File: wrappers/go/rsa.go.in (hunk 1 above)
Comment (AlessandroBudroni) [inline wrappers/go/rsa.go.in]: Ok!

File: wrappers/go/rsa.go.in (hunk 2 above)
Comment (AlessandroBudroni) [inline wrappers/go/rsa.go.in]: Ok!

**Pull request merged**